lib/meadow/pipeline/actions/create_transcode_job.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Meadow.Pipeline.Actions.CreateTranscodeJob do
+  @moduledoc """
+  Action to create an AWS Elemental MediaConvert Job for an audio or video file
+  """
+
+  @mediaconvert_client Application.compile_env(:meadow, :mediaconvert_client)
+
+  alias Meadow.Data.{ActionStates, FileSets}
+  alias MediaConvert.{AudioTemplate, VideoTemplate}
+  alias Sequins.Pipeline.Action
+
+  use Action
+  use Meadow.Pipeline.Actions.Common
+
+  require Logger
+
+  @actiondoc "Create an AWS Elemental MediaConvert Job for an audio or video FileSet"
+
+  defp already_complete?(file_set, _) do
+    ActionStates.ok?(file_set.id, __MODULE__)
+  end
+
+  defp process(file_set, attributes, _false) do
+    ActionStates.set_state!(file_set, __MODULE__, "started")
+
+    source = file_set.core_metadata.location
+    destination = FileSets.streaming_uri_for(file_set)
+    mime_type = file_set.core_metadata.mime_type
+    user_metadata = Map.put(attributes, :file_set_id, file_set.id)
+
+    if supported_mime_type?(mime_type) do
+      case transcode(user_metadata, mime_type, source, destination) do
+        {:ok, job_id} ->
+          Logger.info("MediaConvert Job #{job_id} created")
+          ActionStates.set_state!(file_set, __MODULE__, "ok")
+          :ok
+
+        {:error, error} ->
+          ActionStates.set_state!(file_set, __MODULE__, "error", error)
+          {:error, error}
+      end
+    else
+      ActionStates.set_state!(file_set, __MODULE__, "n/a")
+      :ok
```

Comment:
If you change this return value to `:skip` instead of `:ok`, we can use that as a subscription filter for the pending `TranscodeComplete` action.
```suggestion
      :skip
```